rust/pact_ffi/src/mock_server/mod.rs

```diff
@@ -183,192 +212,267 @@ ffi_fn! {
   }
 }
 
-/// External interface to check if a mock server has matched all its requests. The port number is
-/// passed in, and if all requests have been matched, true is returned. False is returned if there
-/// is no mock server on the given port, or if any request has not been successfully matched, or
-/// the method panics.
-#[no_mangle]
-pub extern fn pactffi_mock_server_matched(mock_server_port: i32) -> bool {
-  let result = catch_unwind(|| {
-    pact_mock_server::mock_server_matched(mock_server_port)
-  });
-
-  match result {
-    Ok(val) => val,
-    Err(cause) => {
-      error!("Caught a general panic: {:?}", cause);
-      false
+ffi_fn! {
+    /// External interface to check if a mock server has matched all its requests. The port number is
+    /// passed in, and if all requests have been matched, true is returned. False is returned if there
+    /// is no mock server on the given port, or if any request has not been successfully matched, or
+    /// the method panics.
+    #[tracing::instrument(level = "trace")]
+    fn pactffi_mock_server_matched(mock_server_port: i32) -> bool {
```

Comment:
I think we still need the catch_unwinds here, because, for instance, this line
```
let mut guard = MANAGER.lock().unwrap();
```
will panic if the mutex has been poisoned, and panics across FFI boundaries result in undefined behaviour.